packages/emotion-theming/src/theme-provider.js

```diff
@@ -1,19 +1,27 @@
+// @flow
 // adapted from styled-components' ThemeProvider
 // https://github.com/styled-components/styled-components/blob/4503cab5b86aa9ef8314c5baa360a2fbb4812485/src/models/ThemeProvider.js
 
-import React, { Component } from 'react'
+import { Component, Children } from 'react'
 import createBroadcast from './create-broadcast'
-import { channel, contextTypes } from './utils'
+import { channel, contextTypes, type Theme } from './utils'
 
 const isPlainObject = test =>
   Object.prototype.toString.call(test) === '[object Object]'
 
-class ThemeProvider extends Component {
+type Props = {
+  theme: Theme
+}
+
+class ThemeProvider extends Component<Props> {
+  getTheme: ThemeProvider.prototype.getTheme
   constructor() {
     super()
     this.getTheme = this.getTheme.bind(this)
   }
-
+  outerTheme: Object
+  broadcast: *
+  unsubscribeToOuterId: number
```

Comment:
this is not entirely true, the correct type would be `?number` although making it so would probably cause flow complain before using this and would force a runtime check 🤔